### ISSUE TITLE
False positive and false negative for alias-path-traversal

### ISSUE THREAD

**Issue by msorens** [returntocorp/sgrep-rules#1159]
**Describe the bug**

Items marked in green are correct; red are incorrect.

Rule: generic.nginx.security.alias-path-traversal

 - False negative on (2).
 - False positive on (3).
 - The failure count at (4) is incorrect; it should say 2 tests failed.
	
<img width="1080" alt="image" src="https://user-images.githubusercontent.com/6817500/110899762-87446f80-82b6-11eb-81cd-d211090644f9.png">

**To Reproduce**
https://semgrep.dev/s/msorens:alias-path-traversal-errors??registry=generic.nginx.security.alias-path-traversal

**Expected behavior**
 - Item (2) should have been flagged by the rule definition.
 - Item (3) should not have been flagged.
 - Item (4) should say 2 tests failed.

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [x] P2: annoying but not blocking me

**Screenshots**

**Desktop (please complete the following information):**
 - OS: macOs
 - Browser Chrome
 - Version 88

**Additional context**
NA

**Comment by msorens** [comment]
Besides the issues noted above, which probably have repercussions beyond just this rule, I do not believe the rule itself is valid. I am not well-versed in nginx configuration but what I see in the configs used in my repo include these which seem valid enough:
```
location ~* ^/e/[^/]+/?(.*)$ {

location ~ "^/organizations/([^/]+)/required_recipe$" {

location /automate.conf.js {
```
